benchmark/benchmark_umf.hpp

```diff
@@ -155,6 +156,60 @@ struct os_provider : public provider_interface {
     static std::string name() { return "os_provider"; }
 };
 
+struct fixed_provider : public provider_interface {
+    void *mem = NULL;
+    const size_t size = 1024 * 1024 * 1024; // 1GB
+    virtual void SetUp([[maybe_unused]] ::benchmark::State &state) override {
+        if (state.thread_index() != 0) {
+            return;
+        }
+
+        if (!mem) {
+            mem = new char[size];
+        }
+
+        provider_interface::SetUp(state);
+    }
+
+    virtual void TearDown([[maybe_unused]] ::benchmark::State &state) override {
+        if (state.thread_index() != 0) {
+            return;
+        }
+
+        if (mem) {
+            delete[] static_cast<char *>(mem);
+            mem = NULL;
+        }
+
+        provider_interface::TearDown(state);
+    }
+
+    provider_interface::params_ptr
+    getParams(::benchmark::State &state) override {
+        umf_fixed_memory_provider_params_handle_t raw_params = nullptr;
+        umfFixedMemoryProviderParamsCreate(&raw_params, mem, size);
+        if (!raw_params) {
+            state.SkipWithError("Failed to create fixed provider params");
+            return {nullptr, [](void *) {}};
+        }
+
+        // Use a lambda as the custom deleter
+        auto deleter = [](void *p) {
+            auto handle =
+                static_cast<umf_fixed_memory_provider_params_handle_t>(p);
+            umfFixedMemoryProviderParamsDestroy(handle);
+        };
+
+        return {static_cast<void *>(raw_params), deleter};
```

Comment:
You can also get the type of first argument to make it more generic:
`provider_interface::params_ptr::element_type`